thorn/src/kernel/kernel.c

```diff
@@ -57,34 +57,38 @@ void kernel_process () {
     }
 }
 
+void kernel_init (void) {
+    uart_init ();
+    init_printf (0, putc);
+    irq_vector_init ();
+    timer_init ();
+    enable_interrupt_controller ();
+    enable_irq ();
+    task_init ();
+
+    printf ("Initialising Framebuffer...\r\n");
+    int gpu_status = init_gpu ();
+    if (!gpu_status) {
+        printf ("Error while initialising Framebuffer\r\n");
+    } else {
+        color * fb = get_fb ();
+        if (!fb) {
+            printf ("Error: Invalid Framebuffer received\r\n");
+        } else {
+            printf ("Received framebuffer: %p\r\n", fb);
+        }
+    }
+
+    LOG("Initialisation done");
+}
+
 
 void kernel_main (int processor_id) {
 
     static volatile unsigned int current_processor = 0;
 
     if (processor_id == 0) {
-        uart_init ();
-        init_printf (0, putc);
-        irq_vector_init ();
-        timer_init ();
-        enable_interrupt_controller ();
-        enable_irq ();
-        //        task_init ();
-
-        LOG ("Logging works");
-
-        printf ("Initialising Framebuffer...\r\n");
-        int gpu_status = init_gpu ();
-        if (!gpu_status) {
-            printf ("Error while initialising Framebuffer\r\n");
-        } else {
-            color * fb = get_fb ();
-            if (!fb) {
-                printf ("Error: Invalid Framebuffer received\r\n");
-            } else {
-                printf ("Received framebuffer: %p\r\n", fb);
-            }
-        }
+        kernel_init ();
     }
 
     while (processor_id != current_processor)
```

Comment:
Everything that isn't the current processor hangs?

Reply:
is a static variable actually shared between cores?

Reply:
yes

Reply:
a static variable is guaranteed to only ever be initialised once

Reply:
got the code now :+1: